wpilibj/src/main/java/edu/wpi/first/wpilibj/controller/PIDControllerRunner.java

```diff
@@ -28,6 +27,12 @@
   // Controller.update() is already running at that time.
   private final ReentrantLock m_outputMutex = new ReentrantLock();
 
+  // This is declared after all other member variables so that during
+  // PIDControllerRunner destruction, the Notifier is stopped before any member
+  // variables its callable uses are destructed. This avoids use-after-free
+  // bugs like crashes when locking is attempted on deallocated mutexes.
+  private final Notifier m_notifier = new Notifier(this::run);
```

Comment:
Is this comment true for Java?

Reply:
I don't know how Java does its destruction. As far as I know, it just never calls the destructor, so what we do here doesn't matter anyway. Putting Notifier after is more correct anyway though.